packages/http-client/src/http-client.ts

```diff
@@ -121,6 +122,31 @@ export function execute<ReturnT>(executeFn: ExecuteHttpRequestFn<ReturnT>) {
   };
 }
 
+function encodeRequestQueryParameters(
+  parameter: Record<string, string> | undefined
+): Record<string, string> | undefined {
+  if (parameter) {
+    return Object.fromEntries(
+      Object.entries(parameter).map(([key, value]) => [
+        encodeURIComponent(key),
+        encodeURIComponent(parameter[key])
+      ])
+    );
+  }
+}
+
+function getEncodedParameters(
+  parameters: OriginOptionsInternal
+): OriginOptionsInternal {
+  return {
+    ...parameters, // sdk parameters are encoded and custom parameters should be taken as they are
```

Comment:
As SDK developer, we use `requestConfig` for SDK parameters.
However, other users (like CAP and the issue author) might miss the encoding, as SDK will not handle it for them.

Reply:
what do you think?

Reply:
Nice alternatives, that worth a separate PR.
However, without that PR, we might want to align with CAP/users about the current situation, because this PR might no solve the original issue.
We can also handle this PR now and release it together with this PR.

Reply:
I think I will include it in this PR. The changes should be not too much. Here just two files are changed.